### PR TITLE
Add some clarity to the documentation on pauseWhen

### PR DESCRIPTION
**Pull request by barryoneill**
I found the existing documentation very confusing - as did someone else https://gitter.im/functional-streams-for-scala/fs2?at=5dd6afa31bf5192e667c4129

### CODE REVIEW

**Comment by barryoneill** [comment]
Not sure why a doc change would trigger: 

```
2020-07-09T01:43:38.1452333Z [info]   handleErrorWith
2020-07-09T01:43:38.1519581Z [info]   - 1 (4 milliseconds)
2020-07-09T01:43:38.1577208Z [info]   - 2 (1 millisecond)
2020-07-09T01:43:38.1592449Z [info]   - 3 (1 millisecond)
2020-07-09T01:43:38.1614316Z [info]   - 4 - error in eval (1 millisecond)
2020-07-09T01:43:38.1634978Z [info]   - 5 (1 millisecond)
2020-07-09T01:43:38.1652641Z [info]   - 6 (1 millisecond)
2020-07-09T01:43:38.1701312Z [info]   - 7 - parJoin (4 milliseconds)
2020-07-09T01:43:38.1724615Z [info]   - 8 (2 milliseconds)
2020-07-09T01:43:38.1771326Z [info]   - 9 (2 milliseconds)
2020-07-09T01:43:38.1833399Z [info]   - 10 (2 milliseconds)
2020-07-09T01:43:38.1854006Z [info]   - 11 (1 millisecond)
2020-07-09T01:43:38.1883213Z [info]   - 12 (2 milliseconds)
2020-07-09T01:43:38.1922263Z [info]   - 13 (1 millisecond)
2020-07-09T01:43:38.1939790Z [info]   - 14 (3 milliseconds)
2020-07-09T01:43:38.2002084Z [info]   - 15 (4 milliseconds)
2020-07-09T01:43:38.2689976Z [info]   - 16 - parJoin CompositeFailure *** FAILED *** (36 milliseconds)
2020-07-09T01:43:38.2690712Z [info]     2 did not equal 3 (StreamSpec.scala:1563)
2020-07-09T01:43:38.2691049Z [info]     org.scalatest.exceptions.TestFailedException:
2020-07-09T01:43:38.2691369Z [info]     ...
2020-07-09T01:43:38.2691680Z [info]     at fs2.StreamSpec.$anonfun$new$712(StreamSpec.scala:1563)
2020-07-09T01:43:38.2691993Z [info]     at fs2.Fs2Spec$Asserting.$anonfun$asserting$2(Fs2Spec.scala:98)
2020-07-09T01:43:38.2692347Z [info]     at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:87)
2020-07-09T01:43:38.2692689Z [info]     at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:368)
2020-07-09T01:43:38.2693036Z [info]     at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:387)
2020-07-09T01:43:38.2693361Z [info]     at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:330)
2020-07-09T01:43:38.2693685Z [info]     at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:103)
2020-07-09T01:43:38.2693994Z [info]     ...
```

**Comment by mpilquist** [comment]
Thanks - that parJoin test fails intermittently. We should really disable it or fix it.

File: core/shared/src/main/scala/fs2/Stream.scala [hunk 1]
```diff
@@ -2309,7 +2309,7 @@ final class Stream[+F[_], +O] private[fs2] (private val free: FreeC[F, O, Unit])
   )(f: (O2, O3) => O4): Stream[F2, O4] =
     this.parZip(that).map(f.tupled)
 
-  /** Like `interrupt` but resumes the stream when left branch goes to true. */
+  /** Pause this stream when `pauseWhenTrue` emits `False`, resuming when `True` is emitted. */
```
Comment (kubukoz) [inline core/shared/src/main/scala/fs2/Stream.scala:2312]: Hmm, I think this is actually the opposite:

```scala
@ val process = Stream(false) ++ Stream.sleep_(2.seconds) ++ Stream(true) ++ Stream.sleep_(5.seconds) ++ Stream(false)
process: Stream[IO[x], Boolean] = Stream(..)

@ Stream.awakeEvery[IO](1.second).debug().pauseWhen(process.debug()).compile.drain.timeout(10.seconds).unsafeRunSync()
false
1006586025 nanoseconds
true
2006826076 nanoseconds
<long delay here>
false
7002943065 nanoseconds
```

So it looks to me that the stream is indeed paused when `pauseWhenTrue` emits true. Which makes sense, given its name :)

Comment (kubukoz) [reply core/shared/src/main/scala/fs2/Stream.scala:2312]: #1949

**Pull request merged**